examples/ex02_cksum/test/ChecksumRTL_test.py

```diff
@@ -116,14 +117,15 @@ def setup_method( s, method ):
   def teardown_method( s, method ):
     s.current_test_method_name = ""
 
-  def run_sim( s, th, max_cycles=1000 ):
+  def run_sim( s, th ):
 
-    s.vcd_file_name = s.__class__.cmdline_opts["dump_vcd"]
+    vcd_file_name = s.__class__.cmdline_opts["dump_vcd"]
+    max_cycles = s.__class__.cmdline_opts["max_cycles"]
 
     # Check for vcd dumping
-    if s.vcd_file_name:
-      s.set_metadata( VcdGenerationPass.vcd_file_name, s.vcd_file_name )
-      s.set_metadata( PrintTextWavePass.enable, True )
+    if vcd_file_name:
+      th.set_metadata( VcdGenerationPass.vcd_file_name, vcd_file_name )
+      th.set_metadata( PrintTextWavePass.enable, True )
 
     # Create a simulator
     th.apply( DefaultPassGroup() )
```

Comment:
Do you think it's possible to also use `test_util.run_sim` here?

Reply:
sure, if we remove the textwave option, we can use run_sim